added some global defaults

tking53 opened the pull request:
1) Enabled global and module specific traceDelays in addition to the per channel ones, as well as reverted the units to ns (from samples). I do a conversion based on the frequency, this makes the value we put into Pixie  and whats in the config file the same base unit (seconds). This is how it was done before. I believe this will reduce some headache when making config files during experiments

2) Enabled Global Firmware and Frequency for use in nonmixed setups. 

I put both of these in the highest <Map> line; and the TraceDelay is required here, the Firm/Freq is only required to be set either here or in the module. The Module line overrides the global for that module, and similarly the per channel value overrides the module and global. This ensures that they are at least something for every module. 

I also changed when the MapNodeParser enters the channel node child parsers, so that the Trace and Fit child nodes are always set if the channel is a Vandle type since the defaults are for vandle:medium anyways. Again if it exists it should override the default behavior.

All of these changes make the transition to the new Dev style much easier, as well as keeping the config file from growing to monstrous  for full experiments. 

I have been using these for the last couple days without issue, with 17N data as well as the ORNL2016 data (which requires that the first module have a different Trace Delay than the rest)

tking53 committed:
added some global defaults

Enabled global and module specific traceDelays in addition to the per channel ones. 

Enabled Global Firmware and Frequency for nonmixed setups.
I also changed the what the defaultChannelParms are set so that even if the Trace and Fit sub nodes arent there AND its for a vandle it sets the defaults since they are for vandle:medium anyways

tking53 commented:
here is what the beginning of my Map node looks like for ORNL2016 data; in case i wasn't clear in my explanation above.
```
    <Map verbose_calibration="False" verbose_map="False" verbose_walk="False" TraceDelay="264" firmware="34300" frequency="250">
      <Module number="0" TraceDelay="188">
        <Channel number="0" type="beta" subtype="double" tags="start,left">
          <Trace RangeLow="15" RangeHigh="5"/>
          <Fit beta="0.004265" gamma="0.1544"/>
        </Channel>
        <Channel number="1" type="beta" subtype="double" tags="start,right">
          <Trace RangeLow="15" RangeHigh="5"/>
          <Fit beta="0.004265" gamma="0.1544"/>
        </Channel>
```

akeeler reviewed (changes requested):
if map trace delay is not set, make error appear in a more accessible place.

tking53 committed:
Making Requested changes for GlobalDefaults

I changed errors from an invalid_argument to IOexceptions so that the error will be easier to find in the parsing output.
I also fixed a few style things that Clion suggested.
